### PR TITLE
Handling .Net wrapping with Swig

### PR DESCRIPTION
**Pull request by mpoudret**

### CODE REVIEW

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Included in ../src/CmakeLists.txt if Phyton build is selected.
+# Included in ../CmakeLists.txt if Phyton build is selected.
```
Comment (philippeVerney) [inline python/CMakeLists.txt]: typo on python

File: cs/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+endif (SWIG_LINKED_TO_RELEASE)
+execute_process(COMMAND powershell  ${EXECUTE_COMMAND})
+message("SWIG C# files have been generated.")
+
+#Add the swig cpp file to the FesapiCpp
+target_sources(${LIBRARY_NAME} PRIVATE ${CMAKE_SOURCE_DIR}/swig/swigGeneratedCsWrapper.cpp)
+
+# Create the Visual Studio project from a template file
+if (SWIG_LINKED_TO_RELEASE)
+	set (CS_LIBRARY_NAME FetpapiCs)
```
Comment (philippeVerney) [inline cs/CMakeLists.txt]: no difference with the "else" I think

File: cs/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+if (WITH_RESQML2_2)
+	string(APPEND SWIG_DEFINITION "-DWITH_RESQML2_2")
+endif (WITH_RESQML2_2)
+if (WITH_ETP)
+	string(APPEND SWIG_DEFINITION " -DWITH_ETP")
+endif (WITH_ETP)
+
+# SWIG execution
+if (SWIG_LINKED_TO_RELEASE)
+	set ( EXECUTE_COMMAND "${SWIG_EXECUTABLE} ${SWIG_DEFINITION} -v -c++ -csharp -dllimport ${LIBRARY_NAME}.dll -namespace F2iConsulting.Fetpapi -o ${CMAKE_SOURCE_DIR}/swig/swigGeneratedCsWrapper.cpp -outdir ${CMAKE_SOURCE_DIR}/cs/src ${CMAKE_SOURCE_DIR}/swig/swigEtp1_2Include.i")
```
Comment (philippeVerney) [inline cs/CMakeLists.txt]: no difference with the "else" I think

File: cmake/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+// Les informations générales relatives à un assembly dépendent de 
+// l'ensemble d'attributs suivant. Changez les valeurs de ces attributs pour modifier les informations
+// associées à un assembly.
+[assembly: AssemblyTitle("FETPAPI CSharp")]
+[assembly: AssemblyDescription("")]
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("F2I-CONSULTING")]
+[assembly: AssemblyProduct("FetpapiCs")]
+[assembly: AssemblyCopyright("Copyright © 2020")]
```
Comment (philippeVerney) [inline cmake/AssemblyInfo.cs]: 2021

File: cs/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+# Cleaning
+execute_process(COMMAND powershell "Remove-Item ${CMAKE_SOURCE_DIR}/cs/src/* -recurse -exclude .gitignore")
+
+set (SWIG_DEFINITION "")
+if (WITH_RESQML2_2)
+	string(APPEND SWIG_DEFINITION "-DWITH_RESQML2_2")
+endif (WITH_RESQML2_2)
+if (WITH_ETP)
+	string(APPEND SWIG_DEFINITION " -DWITH_ETP")
```
Comment (philippeVerney) [inline cs/CMakeLists.txt]: Is it really possible not to be WITH_ETP?

**Review by philippeVerney — CHANGES_REQUESTED** [review]

File: python/CMakeLists.txt (hunk 1 above)
Comment (mpoudret) [reply python/CMakeLists.txt]: Done.

File: cs/CMakeLists.txt (hunk 2 above)
Comment (mpoudret) [reply cs/CMakeLists.txt]: Sure.

File: cs/CMakeLists.txt (hunk 3 above)
Comment (mpoudret) [reply cs/CMakeLists.txt]: Sure.

File: cmake/AssemblyInfo.cs (hunk 4 above)
Comment (mpoudret) [reply cmake/AssemblyInfo.cs]: Done.

File: cs/CMakeLists.txt (hunk 5 above)
Comment (mpoudret) [reply cs/CMakeLists.txt]: Of course not. "-DWITH_RESQML2_2" is not used as well.

**Review by philippeVerney — APPROVED** [review]

**Pull request merged**